benchmark/serializer.py

```diff
@@ -46,16 +47,36 @@ def serialize(dataset_name, dataset_path, serialized_graph_path):
     serialize_queries = [q.strip().replace('{}', dataset_path)
                          for q in serialize_queries]
 
+    table_types = {}
+
     for s in serialize_queries:
         logging.info('Executing query: %s', s)
-        try:
-            # Run kuzu shell one query at a time. This ensures a new process is
-            # created for each query to avoid memory leaks.
-            subprocess.run([kuzu_exec_path, serialized_graph_path],
-                           input=(s + ";" + "\n").encode("ascii"), check=True)
-        except subprocess.CalledProcessError as e:
+        # Run kuzu shell one query at a time. This ensures a new process is
+        # created for each query to avoid memory leaks.
+        process = subprocess.Popen([kuzu_exec_path, serialized_graph_path],
+            stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+        process.stdin.write((s + ";" + "\n").encode("ascii"))
+        process.stdin.close()
+        match = re.match('create\s+(.+?)\s+table\s+(.+?)\s*\(', s, re.IGNORECASE)
+        if match:
+            table_types[match.group(2)] = match.group(1).lower()
+            match = False
```

Comment:
This logic is not very readable. 
1. You should run the check for `create` statement before `subprocess.Popen`. If it is a `create` you should simply pass the input via `stdout` and `stderr` redirection instead of manually get the output and print.
2.  Instead of setting `match` to `False` here you should indent line 66-75 so it is under your `else`. If you do this, you can also avoid the `if` check for your `write` and `flush`.
3. Do not reuse your flags. Call it `create_match` and `copy_match` maybe. Reusing a flag-type variable is 
error-prone and make the code hard to understand.